webapp/views.py

```diff
@@ -767,6 +767,116 @@ def post_advantage_subscriptions(preview):
     return flask.jsonify(purchase), 200
 
 
+def cancel_advantage_subscriptions():
+    api_url = flask.current_app.config["CONTRACTS_LIVE_API_URL"]
+
+    if flask.request.args.get("test_backend", False):
+        api_url = flask.current_app.config["CONTRACTS_TEST_API_URL"]
+
+    user_token = flask.session.get("authentication_token")
+
+    if user_info(flask.session):
+        advantage = AdvantageContracts(
+            session,
+            user_token,
+            token_type=("Macaroon" if user_token else "Bearer"),
+            api_url=api_url,
+        )
+    else:
+        return flask.jsonify({"error": "authentication required"}), 401
+
+    payload = flask.request.json
+
+    if not (
```

Comment:
The request could just be 
```
{
  account_id: $id,
  previous_purchase_id: $id.
  listings: [$listing1, $listing2, ...]
}
```
?
And to avoid looking up values more than once, you could do:
```python
account_id = payload.get("account_id")
previous_purchase_id = payload.get("previous_purchase_id")
listings = payload.get("listings", [])

if not (account_id and previous_purchase_id and listings):
            return flask.jsonify({"error": "bad request"}), 400
```

Reply:
Done.